### PR TITLE
FEAT: Added User Password Change/Update

### PR DESCRIPTION
**Pull request by Gyakusetsu**
# Changes
- Added User Password Change/Update using the route "/me/:id" with PUT request object containing old_password and new_password elements.
- Modified User Model for old password verification
- Modified me-controller and routes for PUT requests on "/me/".
- Added Error Type INVALID_OLD_PASSWORD

# Fixes
- Fixed Auth Controller Error typo

# Postman
image

### CODE REVIEW

File: src/controllers/me-controller.js [hunk 1]
```diff
@@ -12,6 +13,24 @@ module.exports.getInfo = (req, res) => {
     });
 };
 
+module.exports.setNewPassword = async (req, res) => {
+  try {
+    // get old password
+    const user = await User.getById(req.user.id);
```
Comment (mhackyu) [inline src/controllers/me-controller.js]: `user` variable is not used anywhere in the function. Remove this if not needed.

File: src/routes/me.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ const meController = require('../controllers/me-controller');
 const mw = require('./middlewares/auth');
 
 router.get('/', mw.needsAuth, meController.getInfo);
+router.put('/:id', mw.needsAuth, meController.setNewPassword);
```
Comment (mhackyu) [inline src/routes/me.js]: param `:id` is not used in the controller. 
I suggest that you remove this and change the endpoint to `/change-password` to make the endpoint clear.

Comment (Gyakusetsu) [reply src/routes/me.js]: @mhackyu Fixed on commit ca86affab849dc7f30efe509a6ac58c1bb573798

File: src/controllers/me-controller.js (hunk 1 above)
Comment (Gyakusetsu) [reply src/controllers/me-controller.js]: @mhackyu Fixed on commit ca86affab849dc7f30efe509a6ac58c1bb573798

**Pull request merged**